eras/shelley/impl/src/Cardano/Ledger/Shelley/HardForks.hs

```diff
@@ -71,3 +72,13 @@ bootstrapPhase ::
   ProtVer ->
   Bool
 bootstrapPhase pv = pvMajor pv == natVersion @9
+
+-- | Starting with protocol version 10, we need to check that `ppuCoinsPerUTxOByte` is
+-- non-zero when checking the well-formedness of protocol parameter updates but forgo
+-- this check for earlier versions.
+-- See: https://github.com/IntersectMBO/cardano-ledger/issues/4631
+-- and: https://github.com/IntersectMBO/formal-ledger-specifications/issues/571
+forgoCoinsPerUTxOBytesCheck ::
+  ProtVer ->
+  Bool
+forgoCoinsPerUTxOBytesCheck pv = pvMajor pv < natVersion @10
```

Comment:
Just use `bootstrapPhase` instead of defining a new one
```suggestion
```

Reply:
Ah, right since we don't have the `ppuWellFormed` check pre-Conway. :+1: